legend-engine-xt-nonrelationalStore-mongodb-executionPlan-test/src/main/java/org/finos/legend/engine/plan/execution/stores/mongodb/test/shared/MongoDBCommands.java

```diff
@@ -0,0 +1,151 @@
+import com.mongodb.client.MongoDatabase;
+import org.bson.Document;
+import org.eclipse.collections.api.factory.Maps;
+import org.finos.legend.engine.shared.core.ObjectMapperFactory;
+import org.finos.legend.pure.generated.Root_meta_pure_functions_io_http_URL;
+import org.finos.legend.pure.generated.Root_meta_pure_functions_io_http_URL_Impl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class MongoDBCommands
+{
+
+    public static final String DB_ROOT_USERNAME = "sa";
+    public static final String DB_ROOT_PASSWORD = "sa";
+
+    private static final String DB_USER_DATABASE = "userDatabase";
+    private static final String DB_USER_DB_PERSON_COLLECTION = "person";
+    private static final int MONGO_PORT = 27017;
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoDBCommands.class);
+    public static final Map<String, GenericContainer<MongoDBContainer>> CONTAINERS = Maps.mutable.empty();
+    public static final ObjectMapper OBJECT_MAPPER = ObjectMapperFactory.getNewStandardObjectMapperWithPureProtocolExtensionSupports();
+
+    public static String START_SERVER_FUNCTION = "startMongoDBTestServer_String_1__URL_1_";
+
+    public static Root_meta_pure_functions_io_http_URL startServer(String imageTag)
+    {
+        System.out.println("start server");
+
+        Root_meta_pure_functions_io_http_URL_Impl url = new Root_meta_pure_functions_io_http_URL_Impl("mongoDBUrl");
+        GenericContainer<MongoDBContainer> container = CONTAINERS.computeIfAbsent(imageTag, MongoDBCommands::createContainer);
+        url._host(container.getHost());
+        url._port(container.getMappedPort(MONGO_PORT));
+        url._path("/");
+
+
+        return url;
+    }
+
+    public static String STOP_SERVER_FUNCTION = "stopMongoDBTestServer_String_1__Nil_0_";
+
+    public static void stopServer(String imageTag)
+    {
+        System.out.println("stop server");
+        Optional.ofNullable(CONTAINERS.remove(imageTag)).ifPresent(GenericContainer::stop);
+    }
+
+
+    public static String REQUEST_SERVER_FUNCTION = "requestMongoDBTestServer_String_1__String_1__String_1_";
+
+    public static String request(String imageTag, String json)
+    {
+
+        System.out.println("request server");
+        Integer mongoPort = getPortForRunningContainerImage(imageTag);
+
+        MongoClient adminClient = mongoClientForRootAdminWithStaticUserNamePassword(mongoPort);
+
+
+        MongoDatabase userDatabase = adminClient.getDatabase(DB_USER_DATABASE);
+
+        Document bsonCmd = Document.parse(json);
+        Document document = userDatabase.runCommand(bsonCmd);
+
+        System.out.println(document.toJson());
+
+        return "";
```

Comment:
sure, we want to return empty string?

Reply:
I'm a little confused with this request method, this method has only one usage and the output seems to not be used. The usage of it is in the java compiled version of _function <<access.private>> meta::external::store::mongodb::executionTest::v7::collectTest(tag: String[1]): PureTestCollection[1]_. There the variable _requestOutput_ is not used

Reply:
but we still seem to execute the query - that looks odd - as we will be running it from our executor??

Reply:
Elastic is also execution queries ( inserting test data) via pure, if you check the pure function call

meta::external::store::elasticsearch::executionTest::testCase::startESCluster